mlflow/transformers.py

```diff
@@ -990,3 +1110,291 @@ def from_dict(
+            output_key = "answer"
+            data = self._parse_json_encoded_dict_payload_to_dict(data, "table")
+        elif isinstance(self.pipeline, transformers.TokenClassificationPipeline):
+            output_key = "entity"
+        elif isinstance(self.pipeline, transformers.ConversationalPipeline):
+            output_key = None
+            if not self._conversation:
+                self._conversation = transformers.Conversation()
+            self._conversation.add_user_input(data)
+        else:
+            raise MlflowException(
+                f"The loaded pipeline type {type(self.pipeline).__name__} is "
+                "not enabled for pyfunc predict functionality."
+            )
+
+        if isinstance(self.pipeline, transformers.ConversationalPipeline):
+            conversation_output = self.pipeline(self._conversation)
+            return conversation_output.generated_responses[-1]
+        elif isinstance(data, dict):
+            raw_output = self.pipeline(**data)
+        else:
+            raw_output = self.pipeline(data)
+
+        # Handle the pipeline outputs
+        if isinstance(self.pipeline, transformers.FillMaskPipeline):
+            output = self._parse_list_of_multiple_dicts(raw_output, output_key)
+        elif isinstance(self.pipeline, transformers.ZeroShotClassificationPipeline):
+            interim_output = self._parse_lists_of_dict_to_list_of_str(raw_output, output_key)
+            output = self._parse_list_output_for_multiple_candidate_pipelines(interim_output)
+        elif isinstance(self.pipeline, transformers.TokenClassificationPipeline):
+            output = self._parse_tokenizer_output(raw_output, output_key)
+        else:
+            output = self._parse_lists_of_dict_to_list_of_str(raw_output, output_key)
+
+        pre_sanitize = self._pre_sanitize_reduction(data, output)
+
+        return self._sanitize_output(pre_sanitize)
+
+    def _pre_sanitize_reduction(self, input, output):
+        import transformers
+
+        if isinstance(self.pipeline, transformers.TokenClassificationPipeline):
+            return output
+        else:
+            return output[0] if isinstance(input, str) else output
+
+    def _sanitize_output(self, output):
+        # Some pipelines and their underlying models leave leading or trailing whitespace.
+        # This method removes that whitespace.
+        if isinstance(output, str):
+            return output.strip()
+        elif isinstance(output, list):
+            if isinstance(output[0], str):
+                cleaned = [text.strip() for text in output]
+                # If the list has only a single string, return as string.
+                return cleaned if len(cleaned) > 1 else cleaned[0]
+            else:
+                return [self._sanitize_output(coll) for coll in output]
+        elif isinstance(output, dict):
+            return {k: v.strip() for k, v in output.items()}
```

Comment:
Is it safe to always assume that the values of the output dictionaries are strings? if they aren't, this will throw a pretty obscure / hard to parse error message. Do we want to skip sanitization if the keys aren't strings? Seems reasonable to me.

Reply:
added logic to validate contents of both keys and values are strings. If not, added else block to return the output as-is